projects/plugins/jetpack/extensions/blocks/subscriptions/view.js

```diff
@@ -14,33 +14,33 @@ domReady( function () {
 	if ( ! form.payments_attached ) {
 		form.payments_attached = true;
 		form.addEventListener( 'submit', function ( event ) {
+			event.preventDefault();
+
 			const email = form.querySelector( 'input[type=email]' ).value;
-			const email_clause = email ? `&email=${ encodeURIComponent( email ) }` : '';
-			const post_id = form.querySelector( 'input[name=post_id]' )?.value;
-			const post_id_clause = post_id ? `&post_id=${ post_id }` : '';
-			const tier_id = form.querySelector( 'input[name=tier_id]' )?.value;
-			const tier_id_clause = tier_id ? `&tier_id=${ tier_id }` : '';
 
 			if ( form.resubmitted || ! email ) {
 				return;
 			}
-			event.preventDefault();
 
-			const url =
-				'https://subscribe.wordpress.com/memberships/?' +
-				'blog=' +
-				form.dataset.blog +
-				'&plan=newsletter' +
-				'&source=jetpack_subscribe' +
-				'&post_access_level=' +
-				form.dataset.post_access_level +
-				'&display=alternate' +
-				post_id_clause +
-				tier_id_clause +
-				email_clause;
+			const post_id = form.querySelector( 'input[name=post_id]' )?.value ?? '';
+			const tier_id = form.querySelector( 'input[name=tier_id]' )?.value ?? '';
+
+			const params = new URLSearchParams( {
+				email: encodeURIComponent( email ),
+				post_id,
+				tier_id,
```

Comment:
I checked on the back-end, and that works fine. I was skeptical if `null` and '' values were handled the same way. 
It seems like it: fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Sfhofpevor%2Qgb%2Qoybtf.cuc%3Se%3Qq4pqnn54%23216%2Q232-og

Reply:
Nice, thanks! Empty strings are little less prone to cause accidental problems e.g. with `empty()` checks rather than string-nulls.

Reply:
>  e.g. with empty() checks rather than string-nulls.

Of course. But you never know how the backend was built until you check it...